src/partner/ducks/CurrentEventViewReducer.tsx

```diff
@@ -0,0 +1,115 @@
+import { IEvent, event } from "src/partner/models/Event";
+
+export interface IState {
+  events: IEvent[];
+  isLoading: boolean;
+  error: any;
+  currentEvent: IEvent;
+  cancelEvent: boolean;
+  openModal: boolean;
+  editEvent: boolean;
+}
+
+export interface IActions {
+  type: String;
+  payload?: any;
+  error?: boolean;
+  meta?: any;
+}
+
+export const initialState: IState = {
+  events: [],
+  isLoading: false,
+  error: undefined,
+  currentEvent: event(),
+  cancelEvent: false,
+  openModal: false,
+  editEvent: false,
+};
+
+export const reducer = (state: IState, action: IActions): IState => {
+  switch (action.type) {
+    case "CANCEL_EVENT":
```

Comment:
favor `constants` over `strings` and follow `ducks` convention for naming `events`, [see](https://github.com/erikras/ducks-modular-redux)

```js
const CANCEL_EVENT = 'poc-chef-client/current-event/CANCEL';
```